File: designPattern/Composite/Basic/src/Directory.java
import java.util.ArrayList;
import java.util.Iterator;

public class Directory extends Entry{
    private String name;
    private int size = 0;
    private ArrayList documents = new ArrayList();

    public Directory(String name){
        this.name = name;
    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public int getSize() {
        Iterator iterator = documents.iterator();
        while(iterator.hasNext()){
            Entry entry = (Entry)iterator.next();
            size+=entry.getSize();
        }
        return size;
    }

    @Override
    public void printLine(String prefix) {
        System.out.println(prefix+"/"+this);
        Iterator iterator = documents.iterator();
        while(iterator.hasNext()){
            Entry entry = (Entry) iterator.next();
            entry.printLine(prefix+"/"+entry.getName());
        }
    }

    @Override
    public Entry add(Entry entry) {
        documents.add(entry);
        return this;
    }
}

File: designPattern/Composite/basicPython/Directory.py
from Entry import Entry


class Directory(Entry):

    def __init__(self, name):
        self.name = name
        self.dirs = []

    def getName(self):
        return self.name

    def getSize(self):
        size = 0
        for dir_or_file in self.dirs:
            size += dir_or_file.getSize()
        return size

    def add(self, dir_or_file):
        self.dirs.append(dir_or_file)

    def print_line(self, prefix):
        print(prefix+"/"+self.getInformation())
        for dir_or_file in self.dirs:
            dir_or_file.print_line(prefix+"/"+self.name)
